### ISSUE TITLE
add Annotation support to fluent-operator Deployment

### ISSUE THREAD

**Issue by momoXD007** [fluent/fluent-operator#456]
### Is your feature request related to a problem? Please describe.

I am frustrated because I can not add an annotation to fluent-operator Deployment or Pods. There is no way in the templates provided to add annotations.

### Describe the solution you'd like

Basically I just want annotations to be considered in fluent-operator-deployment.yaml and an option to overwrite annotations via  values.yaml
A similar think is already implemented for fluent-bit.

### Additional context

Many people rely on annotations in kubernetes to supply context. Annotations are used to control prometheus behaviour towards this pod or might be used to tell a security-provider in a cluster that a certain pod can or can not do certain things. 

In short supporting annotations is a good practice and large scale production environments will need them.

**Comment by momoXD007** [comment]
Looking at the source-code and templates again: I think it should be seriously considered to support arbitrary labels for fluent-bits fluentds and fluent-operator as well.
I believe for fluent-operator this is can possibly be solved purely in the templates. For fluent-bit I guess template and fluentbit-types.yaml need enhancement


For fluentd I did not check but probably same approach as fluent-bit should work.

**Comment by squaricdot** [comment]
this can be closed right
